src/dbnode/storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -788,6 +798,26 @@ func (s *fileSystemSource) read(
 			setOrMergeResult(r.result)
 		}
 		logSpan("bootstrap_from_index_persisted_blocks_done")
+
+		logSpan("bootstrap_from_index_snapshots_start")
+		// NB(bodu): We don't actually bootstrap snapshot data in the fs bootstrapper
+		// (we do this in the commitlog bootstrapper) but we do want to subtract the shard time
+		// ranges fulfilled by index snapshots.
+		r, err = s.bootstrapFromIndexSnapshots(
```

Comment:
I don't follow why you need to read index snapshots in both bootstrapers

Reply:
Spoke offline.